Plugins/GitUIPluginInterfaces/ObjectId.cs

```diff
@@ -48,322 +49,164 @@ public static ObjectId Random()
-        /// <param name="index">The index within <paramref name="bytes"/> to commence parsing from.</param>
-        /// <returns>The parsed <see cref="ObjectId"/>.</returns>
-        [MustUseReturnValue]
-        public static ObjectId Parse(byte[] bytes, int index)
-        {
-            return new ObjectId(Read(), Read(), Read(), Read(), Read());
-
-            uint Read() => (uint)((bytes[index++] << 24) |
-                                  (bytes[index++] << 16) |
-                                  (bytes[index++] << 8) |
-                                   bytes[index++]);
-        }
-
-        /// <summary>
-        /// Parses an <see cref="ObjectId"/> from ASCII <paramref name="bytes"/> at the given <paramref name="index"/>.
-        /// </summary>
-        /// <remarks>
-        /// <para>Unlike <see cref="Parse(byte[],int)"/> which reads raw bytes, this method reads human-readable
-        /// ASCII-encoded bytes, which are more verbose. Several git commands emit them in this form.</para>
-        /// <para>For parsing to succeed, there must be 40 bytes in <paramref name="bytes"/> starting at <paramref name="index"/>.</para>
-        /// </remarks>
-        /// <param name="bytes">The byte array to parse from.</param>
-        /// <param name="index">The index within <paramref name="bytes"/> to commence parsing from.</param>
+        /// <param name="array">The char span to parse.</param>
         /// <param name="objectId">The parsed <see cref="ObjectId"/>.</param>
         /// <returns><c>true</c> if parsing succeeded, otherwise <c>false</c>.</returns>
-        [MustUseReturnValue]
-        public static bool TryParseAsciiHexBytes(byte[] bytes, int index, [NotNullWhen(returnValue: true)] out ObjectId? objectId)
+        public static bool TryParse(in ReadOnlySpan<char> array, [NotNullWhen(returnValue: true)] out ObjectId? objectId)
         {
-            if (index < 0 || index > bytes.Length - Sha1CharCount)
-            {
-                objectId = default;
-                return false;
-            }
-
-            return TryParseAsciiHexBytes(
-                new ArraySegment<byte>(bytes, index, Sha1CharCount),
-                out objectId);
-        }
-
-        [MustUseReturnValue]
-        public static bool TryParseAsciiHexBytes(ArraySegment<byte> bytes, int index, [NotNullWhen(returnValue: true)] out ObjectId? objectId)
-        {
-            // TODO get rid of this overload? slice the array segment instead
-
-            if (bytes.Array is null)
+            if (array.Length != Sha1CharCount)
             {
                 objectId = default;
                 return false;
             }
 
-            if (index < 0 || bytes.Offset + index + Sha1CharCount > bytes.Array.Length)
+            if (!uint.TryParse(array[..8], NumberStyles.AllowHexSpecifier, provider: null, out uint i1)
+                || !uint.TryParse(array.Slice(8, 8), NumberStyles.AllowHexSpecifier, provider: null, out uint i2)
```

Comment:
`[..]` vs. `.Slice(,)`?

Reply:
I can keep it consistent, but there is a suggestion to change the code Slice(0,x) to [..x], so I followed that

Reply:
❔
```suggestion
            if (!uint.TryParse(array[..8], NumberStyles.AllowHexSpecifier, provider: null, out uint i1)
                || !uint.TryParse(array[8, 16], NumberStyles.AllowHexSpecifier, provider: null, out uint i2)
```
(Although `Slice(..., 8)` was a little clearer in this usecase.)

Reply:
I prefer Slice here, even if it differs from the first.

In general, just accepting the codeanalyzer settings is the easiest.
In this case I did a small test, running slice/range .Length 100K times. Slice() took around 89 ms and [[..8] around 91 ms. This is about 6 ms improvement for 100K (and 30 ms changing all). I did not expect to see a difference...
Added SuppressMessage[] attribute.

With this, I plan to merge this PR